module/zfs/vdev_disk.c

```diff
@@ -615,6 +617,8 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio,
 		abd_offset += BIO_BI_SIZE(dr->dr_bio[i]);
 		bio_offset += BIO_BI_SIZE(dr->dr_bio[i]);
 	}
+	vdev_disk_t *vd = zio->io_vd->vdev_tsd;
+	atomic_inc(&vd->vd_dr);
```

Comment:
Why not just move this whole thing into vdev_disk_dio_get, so you protect against getting negative, vd_dr. Assuming anyone who calls vdev_disk_dio_get correspondingly calls vdev_disk_dio_put.